l5kit/l5kit/simulation/utils.py

```diff
@@ -74,3 +74,50 @@ def disable_agents(dataset: ChunkedDataset, allowlist: np.ndarray) -> None:
     dataset.agents["yaw"][mask_disable] *= 0
     dataset.agents["extent"][mask_disable] *= 0
     dataset.agents["label_probabilities"][mask_disable] = -1
+
+
+def get_frames_subset(dataset: ChunkedDataset, frame_start_idx: int, frame_end_idx: int) -> ChunkedDataset:
+    """Get a new dataset with frames between start (included) and end (excluded).
+    Assumptions:
+        the dataset has only 1 scene
+        the dataset is in numpy format and not zarr anymore
+
+    :param dataset: the single-scene dataset.
+    :param frame_start_idx: first frame to keep.
+    :param frame_end_idx: where to stop taking frames (excluded).
+
+    """
+    if not len(dataset.scenes) == 1:
+        raise ValueError(f"dataset should have a single scene, got {len(dataset.scenes)}")
+    if not isinstance(dataset.agents, np.ndarray):
+        raise ValueError("dataset agents should be an editable np array")
+    if not isinstance(dataset.tl_faces, np.ndarray):
+        raise ValueError("dataset tls should be an editable np array")
+    if not isinstance(dataset.frames, np.ndarray):
+        raise ValueError("dataset frames should be an editable np array")
+    if frame_start_idx >= len(dataset.frames):
+        raise ValueError(f"frame start {frame_start_idx} is over the length of the dataset")
+    if frame_end_idx > len(dataset.frames):
+        raise ValueError(f"frame end {frame_end_idx} is over the length of the dataset")
+    if frame_start_idx >= frame_end_idx:
+        raise ValueError(f"end frame {frame_end_idx} should be higher than start {frame_start_idx}")
+    if frame_start_idx < 0:
+        raise ValueError(f"start frame {frame_start_idx} should be positive")
+
+    new_dataset = ChunkedDataset("")
```

Comment:
Just to understand here, what the `""` mean for the `ChunkedDataset` ?

Reply:
the `ChunkedDataset` has a `path` members which will be used when `open` or `initalise` are called. This is an old logic which we should probably replace.... the argument is required so that's why I'm passing and empty string here

Reply:
Thanks